src/components/workflow/workflow.service.ts

```diff
@@ -0,0 +1,1038 @@
+            node('sg')
+          ]
+        ])
+        .detachDelete('rel')
+        .first();
+
+    } catch (e) {
+      this.logger.warning('could not remove security group from state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // we are using security groups as notification groups for now
+  // attachNotificationGroupToState
+  async attachNotificationGroup(session: ISession, input: GroupState): Promise<void>{
+    try{
+      await this.db
+        .query()
+        .match([
+          [
+            node('token', 'Token', {
+              active: true,
+              value: session.token
+            }),
+            relation('in', '', 'token', {
+              active: true,
+            }),
+            node('requestingUser', 'User', {
+              id: session.userId
+            }),
+            relation('in', '', 'member'),
+            node('sg', 'SecurityGroup', {
+              id: input.securityGroupId
+            }),
+          ],
+          [
+            node('state', 'State', {
+              id: input.stateId
+            })
+          ]
+        ])
+        .merge([
+          [
+            node('state'),
+            relation('out', '', 'notification', {
+              active: true,
+              onEnter: true
+            }),
+            node('sg')
+          ]
+        ])
+        .first();
+
+    } catch (e) {
+      this.logger.warning('could not attach security group to state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // removeNotificationGroupFromState
+  async removeNotificationGroup(session: ISession, input: GroupState): Promise<void>{
+    try{
+      await this.db
+        .query()
+        .match([
+          [
+            node('token', 'Token', {
+              active: true,
+              value: session.token
+            }),
+            relation('in', '', 'token', {
+              active: true,
+            }),
+            node('requestingUser', 'User', {
+              id: session.userId
+            }),
+            relation('in', '', 'member'),
+            node('sg', 'SecurityGroup', {
+              id: input.securityGroupId
+            })
+          ],
+          [
+            node('state', 'State', {
+              id: input.stateId
+            }),
+            relation('out', 'rel', 'notification'),
+            node('sg')
+          ]
+        ])
+        .detachDelete('rel')
+        .first();
+
+    } catch (e) {
+      this.logger.warning('could not remove security group from state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // changeCurrentStateInWorkflow
+  async changeCurrentState(session: ISession, input: ChangeCurrentState): Promise<void>{
+    try{
+      // get current state and workflow
+      const workflow = await this.db
+        .query()
+        .match([
+          [
+            ...matchSession(session),
+            relation('in', '', 'member', {
+              admin: true
```

Comment:
We don't need to check for admin to read a property. You can remove the admin line.